src/mol-gl/shader/postprocessing.frag.ts

```diff
@@ -120,7 +121,20 @@ void main(void) {
         }
     #endif
 
-    // outline needs to be handled after occlusion to keep them clean
+    #ifdef dShadowEnable
+        if (!isBackground(opaqueDepth)) {
+            viewDist = abs(getViewZ(opaqueDepth));
+            fogFactor = smoothstep(uFogNear, uFogFar, viewDist);
+            vec4 shadow = texture2D(tShadows, coords);
+            if (!uTransparentBackground) {
+                color.rgb = mix(mix(vec3(0), uFogColor, fogFactor), color.rgb, shadow.a);
+            } else {
+                color.rgb = mix(vec3(0) * (1.0 - fogFactor), color.rgb, shadow.a);
+            }
+        }
```

Comment:
Do you think we could get significant extra performance by optimizing the shaders? E.g. reducing the number of operations that happen across the board.

Reply:
The whole shadow pp takes <0.2 ms, compared to 2ms for occlusion, which is much slower due to more and incoherent texture fetches. I don't think we are at the point to do sub-millisecond optimizations :)